### PR TITLE
2-Add Jest and RTL

### PR DESCRIPTION
**Pull request by PengchengYuGithub**
Add jest/RTL and a testing example; Update README file.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint"
+    "lint": "next lint",
+    "test": "jest --watch"
```
[the remark this answers is not in the document]
Comment (PengchengYuGithub) [reply package.json:10]: Thx, already changed. Forget this when following Next official instructions. Currently I think there is no need to keep watching testing.

[the remark this answers is not in the document]
Comment (Double-Twelve) [reply package.json]: > Thx, already changed. Forget this when following Next official instructions. Currently I think there is no need to keep watching testing.

By the way, you can add `--coverage` to show test coverage information

[the remark this answers is not in the document]
Comment (PengchengYuGithub) [reply package.json]: Done. Also includes cli options of --colors and --config jest.config.mjs.

[the remark this answers is not in the document]
Comment (Double-Twelve) [reply package.json]: Is `--config jest.config.mjs` required? Jest will automatically detect and use that file for configuration. You don't need to explicitly specify the `--config` flag

[the remark this answers is not in the document]
Comment (PengchengYuGithub) [reply package.json]: Deleted. Jest says config files called jest.config.js|ts|mjs|cjs|json will be discovered automatically.

**Review by tatoMa — APPROVED** [review]
LGTM

**Pull request merged**